packages/editor-ui/src/modules/tags.ts

```diff
@@ -0,0 +1,103 @@
+
+import { ActionContext, Module } from 'vuex';
+import {
+	ITag,
+	ITagsState,
+	IRootState,
+} from '../Interface';
+import { createTag, deleteTag, getTags, updateTag } from '../api/tags';
+import Vue from 'vue';
+
+const module: Module<ITagsState, IRootState> = {
+	namespaced: true,
+	state: {
+		tags: {},
+		isLoading: false,
+		fetchedAll: false,
+		fetchedUsageCount: false,
+	},
+	mutations: {
+		setLoading: (state: ITagsState, isLoading: boolean) => {
+			state.isLoading = isLoading;
+		},
+		setAllTags: (state: ITagsState, tags: ITag[]) => {
+			state.tags = tags
+				.reduce((accu: {[id: string]: ITag}, tag: ITag) => {
+					accu[tag.id] = tag;
+
+					return accu;
+				}, {});
+			state.fetchedAll = true;
+		},
+		upsertTags(state: ITagsState, tags: ITag[]) {
+			tags.map((tag) => {
+				if (state.tags[tag.id]) {
+					const newTag = {
+						...state.tags[tag.id],
+						...tag,
+					};
+					Vue.set(state.tags, tag.id, newTag);
+				}
+				else {
+					Vue.set(state.tags, tag.id, tag);
+				}
+			});
+		},
+		deleteTag(state: ITagsState, id: string) {
+			Vue.delete(state.tags, id);
+		},
+	},
+	getters: {
+		tags(state: ITagsState): ITag[] {
+			return Object.values(state.tags)
+				.sort((a: ITag, b: ITag) => a.name.localeCompare(b.name));
+		},
+		isLoading: (state: ITagsState): boolean => {
+			return state.isLoading;
+		},
+		hasTags: (state: ITagsState): boolean => {
+			return Object.keys(state.tags).length > 0;
+		},
+		getTagById: (state: ITagsState) => {
+			return (id: string) => state.tags[id];
+		},
+	},
+	actions: {
+		fetchAll: async (context: ActionContext<ITagsState, IRootState>, params: {force?: boolean, withUsageCount?: boolean}) => {
+			const {force, withUsageCount} = params || {};
+			if (!force && context.state.fetchedAll && context.state.fetchedUsageCount === !!withUsageCount) {
+				return context.state.tags;
+			}
+
+			context.commit('setLoading', true);
+			const tags = await getTags(context.rootGetters.getRestApiContext, !!withUsageCount);
+			context.commit('setAllTags', tags);
+			context.commit('setLoading', false);
+
+			return tags;
+		},
+		create: async (context: ActionContext<ITagsState, IRootState>, name: string) => {
+			const tag = await createTag(context.rootGetters.getRestApiContext, {name});
+			context.commit('upsertTags', [tag]);
+
+			return tag;
+		},
+		rename: async (context: ActionContext<ITagsState, IRootState>, params: {name: string, id: string}) => {
```

Comment:
```suggestion
		rename: async (context: ActionContext<ITagsState, IRootState>, params: { name: string, id: string }) => {
```

Reply:
Also, we can destructure to slim down, and re-order for consistency.

```suggestion
		rename: async (context: ActionContext<ITagsState, IRootState>, { id, name }: { id: string, name: string }) => {
			const tag = await updateTag(context.rootGetters.getRestApiContext, id, { name });
```